api/server/server.go

```diff
@@ -800,11 +807,14 @@ func (s *Server) postImagesCreate(eng *engine.Engine, version version.Version, w
 		imageImportConfig.ContainerConfig = newConfig
 
 		if err := s.daemon.Repositories().Import(src, repo, tag, imageImportConfig); err != nil {
-			return err
+			if !output.Flushed() {
```

Comment:
Here is the problem. Things are quite similar there for `pull` and `import`.
Assume you wanna pull an non-existing image with current code, you`ll get an error ` http: multiple response.WriteHeader calls`.Seems that any attempt to write to output without check for Flused() would case such problems to my best knowledge.
https://github.com/docker/docker/blob/master/graph/pull.go#L46